### PR TITLE
fix: use all package data

### LINKED ISSUE #15862: Express mode build is needed for a default starter project
### Description of the bug

```
2023-02-07T16:04:06.938+02:00  INFO 45997 --- [  restartedMain] c.v.f.s.frontend.TaskRunDevBundleBuild   : Checking if an express mode bundle build is needed
2023-02-07T16:04:06.952+02:00  INFO 45997 --- [  restartedMain] c.v.f.s.frontend.TaskRunDevBundleBuild   : Dependency @vaadin/bundles is missing from the bundle
2023-02-07T16:04:06.952+02:00  INFO 45997 --- [  restartedMain] c.v.f.s.frontend.TaskRunDevBundleBuild   : Dependency @vaadin/component-base is missing from the bundle
2023-02-07T16:04:06.953+02:00  INFO 45997 --- [  restartedMain] c.v.f.s.frontend.TaskRunDevBundleBuild   : Dependency @vaadin/field-base is missing from the bundle
2023-02-07T16:04:06.953+02:00  INFO 45997 --- [  restartedMain] c.v.f.s.frontend.TaskRunDevBundleBuild   : Dependency @vaadin/input-container is missing from the bundle
2023-02-07T16:04:06.953+02:00  INFO 45997 --- [  restartedMain] c.v.f.s.frontend.TaskRunDevBundleBuild   : Dependency @vaadin/lit-renderer is missing from the bundle
2023-02-07T16:04:06.953+02:00  INFO 45997 --- [  restartedMain] c.v.f.s.frontend.TaskRunDevBundleBuild   : Dependency @vaadin/overlay is missing from the bundle
2023-02-07T16:04:06.953+02:00  INFO 45997 --- [  restartedMain] c.v.f.s.frontend.TaskRunDevBundleBuild   : Dependency @vaadin/vaadin-development-mode-detector is missing from the bundle
2023-02-07T16:04:06.953+02:00  INFO 45997 --- [  restartedMain] c.v.f.s.frontend.TaskRunDevBundleBuild   : Dependency @vaadin/vaadin-usage-statistics is missing from the bundle
2023-02-07T16:04:06.953+02:00  INFO 45997 --- [  restartedMain] c.v.f.s.frontend.TaskRunDevBundleBuild   : An express mode bundle build is needed
```

### Expected behavior

No build is needed

### Minimal reproducible example

```
npx @vaadin/cli init --next hello-v24beta 
cd hello-v24beta
mvn
```

### Versions

- Vaadin / Flow version: 24 beta1
- OS version: mac

### PR DESCRIPTION
**Pull request by caalador**
When validating if bundle contains
required use data for both package.json
and vite handled packages used during
bundle creation.

Add possibility to ignore select packages
known to be used only with frontend
dev-server.

Fixes #15862

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java [hunk 1]
```diff
@@ -400,6 +403,27 @@ private static boolean hashAndBundleModulesEqual(JsonObject statsJson,
...
+     * Collect package.json and vite handled modules used for bundle generation
+     * to compare requirements against. Vite handled modules override any
+     * package.json dependencies as vite versions are more exact.
+     *
+     * @param statsJson
+     *            bundle config stats.json
+     * @return all handled npm pakcages and modules
+     */
+    private static JsonObject collectBuildModules(JsonObject statsJson) {
+        JsonObject bundleModules = statsJson
```
Comment (Artur-) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java:416]:
```suggestion
        JsonObject packageJsonWhenCreatingBundle = statsJson
```

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java [hunk 2]
```diff
@@ -400,6 +403,27 @@ private static boolean hashAndBundleModulesEqual(JsonObject statsJson,
...
+                .getObject("packageJsonDependencies");
+        JsonObject viteDependencies = statsJson.getObject("npmModules");
```
Comment (Artur-) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java:418]:
```suggestion
        JsonObject modulesUsedForBundle = statsJson.getObject("npmModules");
```

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java [hunk 3]
```diff
@@ -493,13 +517,31 @@ public void execute() {
                     .generatePackageJsonHash(packageJson);
             packageJson.getObject(NodeUpdater.VAADIN_DEP_KEY)
                     .put(NodeUpdater.HASH_KEY, hash);
+
+            // Drop un-wanted frontend dev server packages.
+            cleanUnwantedPackages(packageJson);
```
Comment (Artur-) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java:522]: Why should `@vaadin/bundles` be removed from the default package.json? Wouldn't that make the bundle not available in projects?

Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java:522]: This is only an in memory package.json that is used only for validating the requirements of dev-bundle.
It doesn't change the actual package.json in any way so an actual run of npm would contain the package.

**Comment by Artur-** [comment]
This should not be needed thanks to https://github.com/vaadin/platform/pull/3778

**Comment by caalador** [comment]
Closing PR due to fix in default bundle generation.

**Pull request closed without merging**